include/cocaine/hpack/header.hpp

```diff
@@ -21,150 +21,124 @@
 #pragma once
 
 #include <array>
+#include <deque>
 #include <functional>
+#include <string>
 #include <system_error>
 #include <vector>
 
+#include <boost/optional/optional.hpp>
+
 struct ch_header;
 
 namespace cocaine { namespace hpack {
 
 struct init_header_t;
+class header_t;
 
 size_t
 http2_integer_size(size_t sz, size_t bit_offset);
 
-size_t
-http2_integer_encode(unsigned char* dest, uint64_t source, size_t bit_offset, char prefix);
-
 namespace header {
 
-struct data_t {
-    const char* blob;
-    size_t size;
-    bool operator==(const data_t& other) const;
-
-    template<class To>
-    To
-    convert() const {
-        static_assert(std::is_pod<typename std::remove_reference<To>::type>::value &&
-                      !std::is_pointer<typename std::remove_reference<To>::type>::value &&
-                      !std::is_array<typename std::remove_reference<To>::type>::value,
-                      "only POD non pointer, non array data type is allowed to convert header data"
-                      );
-        if(size != sizeof(typename std::remove_reference<To>::type)) {
-            throw std::system_error(std::make_error_code(std::errc::invalid_argument), "invalid header data size");
-        }
-        return *(reinterpret_cast<const To*>(blob));
-    }
-};
-
 template<size_t N>
-inline
-constexpr
-data_t
-create_data(char const (&source)[N]) {
-    return data_t{source, N-1};
+std::string
+binary_pack(char const (&source)[N]) {
```

Comment:
For me it seems as another unnecessary API breakage, really. Moreover the proper way of naming is to `do_what`. For example: `pack_binary`.